File: src/AxisCRM.Api/Domain/Models/Atendimento.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;
using AxisCRM.Api.Domain.Enums;

namespace AxisCRM.Api.Domain.Models
{
    public class Atendimento : EntidadeBase
    {
        public int IdCliente { get; set; }
        public int IdUsuario { get; set; }
        public string Assunto { get; set; }
        public string Historico { get; set; } = $"{DateTime.Now.ToShortDateString()} - Criado atendimento.{Environment.NewLine} ";
        public DateTime? DataEncerramento { get; set; }
        public DateTime? DataUltimaAtualizacao { get; set; }
        public StatusAtendimento Status { get; set; } = StatusAtendimento.Aberto;

        [ForeignKey(nameof(IdCliente))]
        public virtual Cliente Cliente { get; set; }
        
        [ForeignKey(nameof(IdUsuario))]
        public virtual Usuario Usuario { get; set; }
        public virtual ICollection<Parecer> Pareceres { get; set; } = [];
    }
}

File: App/src/models/Atendimento.js
import Parecer from './Parecer';

export default class Atendimento {
  constructor({
    id = 0,
    assunto = '',
    idCliente = 0,
    dataCadastro = null,
    dataEncerramento = null,
    status = 0,
    dataUltimaAtualizacao = null,
    idUsuario = 0,
    pareceres = [],
    historico = ''
  } = {}) {
    this.id = id;
    this.assunto = assunto;
    this.idCliente = idCliente;
    this.dataCadastro = dataCadastro
      ? new Date(dataCadastro)
      : null;
    this.dataEncerramento = dataEncerramento
      ? new Date(dataEncerramento)
      : null;
    this.status = status;
    this.dataUltimaAtualizacao = dataUltimaAtualizacao
      ? new Date(dataUltimaAtualizacao)
      : null;
    this.idUsuario = idUsuario;
    this.pareceres = Array.isArray(pareceres)
      ? pareceres.map(p => new Parecer(p))
      : [];
    this.historico = historico;
  }
}
